### PR TITLE
HPCC-22367 Fix roxie whitelist issue.

### PR DESCRIPTION
**Pull request by jakesmith**
Also add role to "Access denied!" message.
And default to @netAddress if for some reason the named Computer
is not found in the environment.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dali/base/dasess.cpp [hunk 1]
```diff
@@ -565,7 +565,8 @@ class CSessionRequestServer: public Thread
                         INode *na = queryNullNode();
                         Owned<IGroup> dummyCoven = createIGroup(1, &na);
                         dummyCoven->serialize(mb);
-                        Owned<IException> e = makeStringException(666, "Access denied!");
+                        const char *roleName = queryRoleName((DaliClientRole)role);
+                        Owned<IException> e = makeStringExceptionV(666, "Access denied! [role=%s]", roleName);
```
Comment (richardkchapman) [inline dali/base/dasess.cpp]: Is 666 the right number?

Comment (jakesmith) [reply dali/base/dasess.cpp]: There aren't sensible general server client exception in this area (SDS has it's own).
Most places set error code to -1.
666 is misleading (and devilish) - I'll change to -1 for a bit of consistency.

**Comment by richardkchapman** [comment]
@dcamper Can you check this addresses your issue?

**Comment by dcamper** [comment]
@richardkchapman Confirmed that the change works.  I applied this PR as a patch and removed the <WhiteList enabled="false"/> entry I had manually entered into environment.xml.  dalidiag -whitelist now shows RoxieMaster, and ROXIE starts up correctly with the cluster.

**Pull request merged**